### PR TITLE
feat(RN): update getting started pages

### PR DESCRIPTION
**Pull request by kahest**
Updates RN getting started page to use new product selector. 

This also includes several smaller improvements to streamline the page and the navigation:
- move features list to separate new menu item
- remove some superfluous/old notes that are not important enough any more to call out on the first landing page
- move Touch Tracking and Uninstall menu items to Configuration and Troubleshooting, respectively, to clean up the menu structure
- include all the content in docs/platforms/react-native/index.mdx and remove the <PlatformContent> imports

### CODE REVIEW

**Comment by kahest** [comment]
@krystofwoldrich since our [Expo setup page](https://docs.sentry.io/platforms/react-native/manual-setup/expo/) is in effect an almost complete getting started page by itself, I'm thinking we should also apply the new product selector there, wdyt? Or should we create a new guide under RN under Expo, like e.g. [Laravel under PHP](https://docs.sentry.io/platforms/php/guides/laravel/#install)?

File: docs/platforms/react-native/features/index.mdx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+title: Features
+sidebar_order: 1
+description: "Learn more about Sentry's React Native SDK features."
+---
+
 <Note>
 
 Sentry's React Native SDK enables automatic reporting of errors and exceptions, and identifies performance issues in your application. The React Native SDK includes performance monitoring (our `@sentry/tracing` package) by default.
 
 </Note>
```
Comment (krystofwoldrich) [inline docs/platforms/react-native/features/index.mdx:11]: The `@sentry/tracing` package is deprecated over 50 JS feature versions and removed in v8. I think we can remove the note.

File: platform-includes/getting-started-install/react-native.mdx [hunk 2]
```diff
@@ -12,7 +6,7 @@ npx @sentry/wizard@latest -s -i reactNative
 
 <Note>
 
-The wizard doesn't support React Native versions `0.68` and older. To use the wizard with older versions of React Native, you'll have to `confirm` in the wizard dialogue and manually patch the Xcode project later. Read more on the [Manual Configuration](/platforms/react-native/manual-setup/manual-setup/) page.
+The wizard supports React Native versions `0.69` and newer. To use the wizard with older versions of React Native, you'll have to `confirm` in the wizard dialogue and manually patch the Xcode project later. Read more on the [Manual Configuration](/platforms/react-native/manual-setup/manual-setup/) page.
 
 </Note>
```
Comment (krystofwoldrich) [inline platform-includes/getting-started-install/react-native.mdx:11]: The latest supported version of RN by Meta is 0.70, I think we can remove this banner. 

https://reactnative.dev/versions

File: platform-includes/getting-started-install/react-native.mdx [hunk 3]
```diff
@@ -33,8 +27,6 @@ If you're using Expo, [read our docs](/platforms/react-native/manual-setup/expo/
 
 ### iOS Specifics
 
-The SDK needs access to certain information about the device and the application for its essential functionality. Some of the APIs required for this are considered privacy-relevant. Add the privacy manifest to your Xcode project to ensure your app is compliant with Apple's guidelines. Read the [Apple Privacy Manifest](/platforms/react-native/data-management/apple-privacy-manifest/) guide for more info on how to add records required for the Sentry SDKs.
-
 When you use Xcode, you can hook directly into the build process to upload debug symbols and source maps.
 
 ### Android Specifics
```
Comment (krystofwoldrich) [inline platform-includes/getting-started-install/react-native.mdx:32]: Both of these sections can be removed. 

The information about the build phases is included in the wizard description.

The Android info is also outdated `react-native link` command doesn't exist any more, the info is not relevant anymnore.

The `configureondemand` on demand can be moved to troubleshooting.

**Comment by krystofwoldrich** [comment]
I'm not sure if it's against any rules but I think it would be much easier and more readable to include all the content in `docs/platforms/react-native/index.mdx` and remove the `<PlatformContent` imports.

**Comment by krystofwoldrich** [comment]
> apply the new product selector in the Expo page
Yes, I think that makes sense.

> creating a new guide for Expo
As long as the overhead of maintaining the Guide is similar to a page I think it would be clearer for new Sentry users.

**Pull request merged**